core/src/main/java/org/apache/calcite/rel/core/Filter.java

```diff
@@ -139,6 +141,11 @@ public RexNode getCondition() {
     return condition;
   }
 
+  @Override public List<ImmutableBitSet> getInputFieldsUsed() {
```

Comment:
The interpretation of this is open to debate.
All input fields are actually copied to the output.
Maybe this should be documented, and the method should be called getInputFieldsUsedInternally.